File: src/my_project/easy_problems/from201to250/redistribute_characters_to_make_strings_equal.py
from typing import List 

class Solution:
    def makeEqual(self, words: List[str]) -> bool:
    
        temp_dict = dict()
        answer = True    
        vocab = ""
        for word in words:
            vocab = vocab + word
            
        len_vocab: int = len(vocab)
            
        for i in range(len_vocab):
            temp_dict[vocab[i]] = 0
        
        for i in range(len_vocab):
            temp_dict[vocab[i]] += 1
            
        len_words: int = len(words)
        
        for i in range(len_vocab):
            if temp_dict[vocab[i]] % len_words != 0:
                answer = False 
                return answer 
        
        
        return answer  
    
solution = Solution()

print(solution.makeEqual(words = ["abc","aabc","bc"]))

print(solution.makeEqual(words = ["ab","a"]))

File: src/my_project_typescript/easy_problems_typescript/from101to150/redistributeCharactersToMakeStringsEqual.ts
function makeEqual(words: string[]): boolean {

    let answer = true 
    let tempDict: any = {}
    let vocab: string = ""

    for (let word of words){
        vocab = vocab + word
    }

    let lenVocab: number = vocab.length

    for (let i = 0; i < lenVocab; i++){
        tempDict[vocab[i]] = 0
    }

    for (let i = 0; i < lenVocab; i++){
        tempDict[vocab[i]] += 1
    }

    let lenWords: number = words.length

    for (let i = 0; i <lenVocab; i++){
        if (tempDict[vocab[i]] % lenWords !== 0){
            answer = false
            return answer 
        }
    }
    return answer 
};

console.log(makeEqual(["abc","aabc","bc"]))

console.log(makeEqual(["ab","a"]))
